snapcraft/parts/plugins/matter_plugin.py

```diff
@@ -0,0 +1,171 @@
+# -*- Mode:Python; indent-tabs-mode:nil; tab-width:4 -*-
+#
+# Copyright 2023 Canonical Ltd.
+#
+# This program is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License version 3 as
+# published by the Free Software Foundation.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with this program.  If not, see <http://www.gnu.org/licenses/>.
+
+"""The matter plugin."""
+import os
+
+from typing import Any, Dict, List, Set, cast
+
+from craft_parts import infos, plugins
+from overrides import overrides
+
+MATTER_REPO = "https://github.com/project-chip/connectedhomeip.git"
+"""The repository where the matter SDK resides."""
+
+
+class MatterPluginProperties(plugins.PluginProperties, plugins.PluginModel):
+    """The part properties used by the matter plugin."""
+
+    matter_tag: str
+    matter_zap_tag: str
```

Comment:
These could be misleading. The first property is the Matter SDK git references (tags, branches), which differ from Matter standard versions. I suggest adding "sdk" to avoid confusion with Matter standard. Moreover, replace `tag` with `version` to make it less cryptic, since the end user doesn't see that this is used for checking out a git repo.

```suggestion
    matter_sdk_version: str
    matter_zap_version: str
```